src/scripts/slack/notification.go

```diff
@@ -0,0 +1,57 @@
+package slack
+
+import (
+	"log"
+
+	"github.com/CircleCI-Public/slack-orb-go/src/scripts/jsonutils"
+	"github.com/CircleCI-Public/slack-orb-go/src/scripts/stringutils"
+)
+
+type Notification struct {
+	Status                   string
+	Branch                   string
+	Tag                      string
+	Event                    string
+	BranchPattern            string
+	TagPattern               string
+	InvertMatch              bool
+	Template                 string
+	InlineTemplate           string
+	EnvVarContainingTemplate string
+}
+
+func (j *Notification) IsEventMatchingStatus() bool {
+	return j.Status == j.Event || j.Event == "always"
+}
+
+func (j *Notification) IsPostConditionMet() bool {
+	branchMatches, _ := stringutils.IsPatternMatchingString(j.BranchPattern, j.Branch)
+	tagMatches, _ := stringutils.IsPatternMatchingString(j.TagPattern, j.Tag)
+	return (branchMatches || tagMatches) != j.InvertMatch
+
+}
+
+func (j *Notification) BuildMessageBody() string {
```

Comment:
in the same vein as my rambling comment about the slack message, I think returning errors from this function and doing any bailout / stderr printing from the caller likely makes more sense.

Reply:
I added `error` as a return value as well. If there is an error, I `return "", err` and handle the logging in `main`